sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -719,3 +720,99 @@ case class CoGroup(
   override protected def withNewChildrenInternal(
       newLeft: LogicalPlan, newRight: LogicalPlan): CoGroup = copy(left = newLeft, right = newRight)
 }
+
+object JoinWith {
```

Comment:
Can you file an follow-up so we can move this into the analyzer?

Reply:
Added [SPARK-44225](https://issues.apache.org/jira/browse/SPARK-44225)